src/udev/mod.rs

```diff
@@ -13,8 +13,54 @@ use udev::Enumerator;
 
 use self::device::Device;
 
+const RULE_PRIORITY: &str = "59";
 const RULES_PREFIX: &str = "/run/udev/rules.d";
 
+/// Hide all removable input devices from regular users.
+pub async fn block_joysticks() -> Result<(), Box<dyn Error>> {
+    // Find the chmod command to use for hiding
+    let chmod_cmd = if Path::new("/bin/chmod").exists() {
+        "/bin/chmod"
+    } else {
+        "/usr/bin/chmod"
+    };
+
+    let rule = format!(
+        r#"# Hide all evdev devices that are not InputPlumber virtual devices
+ACTION=="add|change", SUBSYSTEM=="input", KERNEL=="js[0-9]*|event[0-9]*", ENV{{ID_INPUT_JOYSTICK}}=="1", ENV{{INPUTPLUMBER_VIRT}}!="1", MODE:="0000", GROUP:="root", RUN:="{chmod_cmd} 000 %p"
```

Comment:
@pastaq We may want to re-add the generic gamepad device definition if we're going to do this.

Reply:
Fair enough.